### PR TITLE
Swift 4 version

### PR DESCRIPTION
**Pull request by sund4nce**
Migrated to Swift 4 for a new project.
Thought it might be useful some others as well. 

Thanks a lot for you contributions & fixes @alehed!

### CODE REVIEW

**Comment by alehed** [comment]
Thanks a lot @mort3m !

This will be merged as soon as the GM Seed comes out. For the meantime you can fix the remaining issues with indentation. As stated in the `.editorconfig` this project uses four spaces for indentation.

File: SyntaxKit/AttributedParser.swift [hunk 1]
```diff
@@ -44,7 +44,7 @@ open class AttributedParser: Parser {
     }
 
     open func attributedString(for string: String, base: Attributes? = nil) -> NSAttributedString {
-        let output = NSMutableAttributedString(string: string, attributes: base)
+		let output = NSMutableAttributedString(string: string, attributes: base)
```
Comment (alehed) [inline SyntaxKit/AttributedParser.swift]: This should be 8 spaces instead of seven.

File: SyntaxKit/Parser.swift [hunk 2]
```diff
@@ -274,7 +274,7 @@ open class Parser {
                     print("Attention unexpected capture (\(index) to \(result.numberOfRanges)): \(expression.pattern)")
                     continue
                 }
-                let range = result.rangeAt(Int(index))
+				let range = result.range(at: Int(index))
```
Comment (alehed) [inline SyntaxKit/Parser.swift]: Same here (missing space).

File: SyntaxKit/Theme.swift [hunk 3]
```diff
@@ -47,14 +47,14 @@ public struct Theme {
 
         var attributes = [String: Attributes]()
         for raw in rawSettings {
-            guard var setting = raw["settings"] as? [String: AnyObject] else { continue }
+            guard var setting = raw["settings"] as? [NSAttributedStringKey: Any] else { continue }
 
-            if let value = setting.removeValue(forKey: "foreground") as? String {
-                setting[NSForegroundColorAttributeName] = Color(hex: value)
+			if let value = setting.removeValue(forKey: NSAttributedStringKey(rawValue: "foreground")) as? String {
```
Comment (alehed) [inline SyntaxKit/Theme.swift]: Also a missing space

File: SyntaxKit/Theme.swift [hunk 4]
```diff
@@ -26,11 +26,11 @@ public struct Theme {
     public let attributes: [String: Attributes]
 
     public var backgroundColor: Color {
-        return attributes[Language.globalScope]?[NSBackgroundColorAttributeName] as? Color ?? Color.white
+		return attributes[Language.globalScope]?[NSAttributedStringKey.backgroundColor] as? Color ?? Color.white
```
Comment (alehed) [inline SyntaxKit/Theme.swift]: Another missing space

File: SyntaxKit/Theme.swift [hunk 5]
```diff
@@ -26,11 +26,11 @@ public struct Theme {
...
     }
 
     public var foregroundColor: Color {
-        return attributes[Language.globalScope]?[NSForegroundColorAttributeName] as? Color ?? Color.black
+		return attributes[Language.globalScope]?[NSAttributedStringKey.foregroundColor] as? Color ?? Color.black
```
Comment (alehed) [inline SyntaxKit/Theme.swift]: Here too.

File: SyntaxKit/Theme.swift [hunk 6]
```diff
@@ -47,14 +47,14 @@ public struct Theme {
...
 
-            if let value = setting.removeValue(forKey: "foreground") as? String {
-                setting[NSForegroundColorAttributeName] = Color(hex: value)
+			if let value = setting.removeValue(forKey: NSAttributedStringKey(rawValue: "foreground")) as? String {
+				setting[NSAttributedStringKey.foregroundColor] = Color(hex: value)
             }
 
-            if let value = setting.removeValue(forKey: "background") as? String {
-                setting[NSBackgroundColorAttributeName] = Color(hex: value)
+			if let value = setting.removeValue(forKey: NSAttributedStringKey(rawValue: "background")) as? String {
```
Comment (alehed) [inline SyntaxKit/Theme.swift]: Last one

**Review by alehed — CHANGES_REQUESTED** [review]
I can't see if the whitespace diff is from using tabs instead of spaces or just indenting with one space less than usual.

**Comment by sund4nce** [comment]
Alright @alehed, I will take care of it!

**Review by alehed — CHANGES_REQUESTED** [review]
Two more lines still contain tabs (53 and 57).

**Review by alehed — APPROVED** [review]

**Comment by alehed** [comment]
Great, thanks! As mentioned, this will be merged as soon as the GM Seed is released by Apple, which is probably going to happen sometime in September.

Unfortunately, there is currently no editor-config plugin that works in Xcode…

**Comment by alehed** [comment]
@mort3m Thanks a lot for the contribution! 👍

**Pull request merged**